src/components/Thought.js

```diff
@@ -435,7 +446,20 @@ const ThoughtContainer = ({
       dragPreview(getEmptyImage())
     }
   }}>
-    <div className='thought-container'>
+    <div className='thought-container'
+      onMouseDown={e => {
```

Comment:
You'll need to handle `onTouchStart` and `onTouchEnd` so that this also works on mobile.

Reply:
I've done research today about drag-and-drop after click/tap and move on text. It looks like we need in dragging process **disable** ContentEditable component and apply style with **user-select: none** to prevent text selection while moving. But after that, event dragEnd fires up after cursor left ContentEditable border, and dragging stops. Need some time to figure out what's happening.

We cant start dragging from click on text because of text selection so we have to disable it. But by disabling it, we prevent text selection, when user wants to select part of the text by mouse. Need to come up with idea how to deal with it. What do you think? Or correct me, if i'm wrong somewhere. Also want to ask, what potential issues or pitfalls i could come across by  disabling ContentEditable?

Reply:
Thanks, Fedor, you helped me realize I missed an important point. Thoughts can be moved by dragging the text of the thought **on mobile only**. On desktop, clicking and dragging text should do the default browser behavior of selecting text within the editable.

So the dragging behavior is great as-is, and you just need a find a way to enable the blue bullet on mobile when the thought is tapped. I think the key will be to re-render the cursor overlay bullet without re-rendering the ContentEditable. You may have to move the cursor overlay bullet to its own component and `connect` it to redux.

Hopefully that is much easier for you! Thanks for being willing to investigate and let me know I'm missing something or you have any other questions.